src/soundvectors.py

```diff
@@ -21,6 +27,16 @@ class Sound(typing.Protocol):  # Sound objects are expected to have a name attri
     name: str
 
 
+class DomainHierarchyType(typing.Protocol):
+    @classmethod
+    def primary_domains(cls) -> typing.Set:
+        pass
+
+    @classmethod
+    def exclude_for_to_sound(cls) -> typing.Set:
+        pass
```

Comment:
Here's the methods expected for a DomainHierarchy implementation - for type hints.